managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/OpAddEntry.java

```diff
@@ -307,7 +307,9 @@ void handleAddFailure(final LedgerHandle lh) {
         finalMl.getExecutor().executeOrdered(finalMl.getName(), SafeRun.safeRun(() -> {
             // Force the creation of a new ledger. Doing it in a background thread to avoid acquiring ML lock
             // from a BK callback.
-            finalMl.ledgerClosed(lh);
+            if (finalMl != null) {
```

Comment:
how can `finalMl` ever be null? Wouldn't `finalMl.getName()` also fail?

Reply:
This looks like a thread safety issue again where the solution isn't to suppress the NPE.

Reply:
or perhaps #12103 already fixed the issue and you are running a version without that fix? @lordcheng10 can you confirm if you have changes for #12103?

Reply:
StreamNative 2.8.1.2 doesn't include #12103: https://github.com/streamnative/pulsar/blob/d952ec78079e48b325310566900ccba7c75ba0e3/managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/OpAddEntry.java#L288-L292